enhancements/microshift/advertise-version-details.md

```diff
@@ -0,0 +1,302 @@
+---
+title: advertise-version-details
+authors:
+  - "@dhellmann"
+reviewers: # Include a comment about what domain expertise a reviewer is expected to bring and what area of the enhancement you expect them to focus on. For example: - "@networkguru, for networking aspects, please look at IP bootstrapping aspect"
+  - "@spadgett, console team"
+  - "@wking, upgrades team"
+  - "@LalatenduMohanty, upgrades team"
+  - "@soltysh, workloads team"
+approvers:
+  - "@sdodson"
+api-approvers: # In case of new or modified APIs or API extensions (CRDs, aggregated apiservers, webhooks, finalizers). If there is no API change, use "None"
+  - None
+creation-date: 2022-07-29
+last-updated: 2022-07-29
+tracking-link: # link to the tracking ticket (for example: Jira Feature or Epic ticket) that corresponds to this enhancement
+  - https://issues.redhat.com/browse/USHIFT-233
+  - https://bugzilla.redhat.com/show_bug.cgi?id=1850656
+---
+
+# Advertise Version Details
+
+## Summary
+
+We have several use cases for allowing authenticated users to discover
+whether they are communicating with OCP or MicroShift, including the
+version of the product. We already have the ClusterVersion resource
+for OCP, but that requires special privileges to read and is not
+present in a MicroShift deployment. This enhancement describes a
+standard that both platforms can use to share platform and version
+information with API callers.
+
+## Motivation
+
+Users configuring applications to deploy on different versions of
+OpenShift may need to adjust their configuration to work differently
+on OCP and MicroShift. Ideally this is not needed, but in practice it
+is useful because of different APIs being available and different
```

Comment:
Capabilities should be discovered not by version but by API discovery which provides information what's available. Version should never be used for that.

Reply:
The goal of exposing that version information should be purely informational, that's what microshift needs and what the linked BZ is describing. Currently CVO required cluster-admin privs which is not something regular users have access to, whereas in parallel all have access to `/version` endpoint exposed by kubernetes.

Reply:
> Capabilities should be discovered not by version but by API discovery which provides information what's available. 

not all cluster behaviors are determined by api availability.  e.g. the recent service account secret change, or the impending pod security admission and default SCC changes.

Reply:
I added details about behavior differences not based on APIs to the body of the document.